test/basics.js

```diff
@@ -338,4 +337,66 @@ describe('Basics', function() {
     done();
   });
 
+  it('should do requestone-get-reply', function(done) {
+    var nc = NATS.connect(PORT);
+    var initMsg = 'Hello World';
+    var replyMsg = 'Hello Back!';
+
+    nc.subscribe('foo', function(msg, reply) {
+      should.exist(msg);
+      msg.should.equal(initMsg);
+      should.exist(reply);
+      reply.should.match(/_INBOX\.*/);
+      nc.publish(reply, replyMsg);
+    });
+
+    var gotOne = false;
+    nc.requestOne('foo', initMsg, null, 1000, function(reply) {
+      should.exist(reply);
+      reply.should.equal(replyMsg);
+      if(! gotOne) {
+        gotOne = true;
+        nc.close();
+        done();
+      }
+    });
+  });
+
+  it('should do requestone-will-unsubscribe', function(done) {
+    var rsub = "x.y.z";
+    var nc = NATS.connect(PORT);
+    nc.subscribe(rsub, function(msg, reply) {
+      reply.should.match(/_INBOX\.*/);
+      nc.publish(reply, "y");
+      nc.flush();
+      setTimeout(function() {
+        nc.publish(reply, "z");
+        nc.flush();
+        nc.close();
+        done();
+      }, 1500);
+    });
+
+    var count = 0;
+    nc.requestOne(rsub, "", null, 1000, function(reply) {
+      reply.should.not.be.instanceof(NATS.NatsError);
+      should.exist(reply);
+      count++;
+      process.nextTick(function() {
+        count.should.equal(1);
+      });
+    });
+  });
+
+
+  it('should do requestone-can-timeout', function(done) {
+    var nc = NATS.connect(PORT);
+    nc.requestOne('a.b.c', '', null, 1000, function(reply) {
+      should.exist(reply);
+      reply.should.be.instanceof(NATS.NatsError);
+      reply.should.have.property('code', NATS.REQ_TIMEOUT);
+      nc.close();
+      done();
+    });
+  });
```

Comment:
I'd also suggest tests that ensure the callback only ever gets fired once for
- multiple responses
- one response, then wait beyond the timeout in the test (as @kozlovic suggested)